Remotion/Core/UnitTests/Context/SafeContextTests/SafeContextTest.Task.ContinueWith.cs

```diff
@@ -0,0 +1,367 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Remotion.Context;
+using static Remotion.UnitTests.Context.SafeContextTests.SafeContextTaskTest.ContinueWithTestFlags;
+
+namespace Remotion.UnitTests.Context.SafeContextTests
+{
+  public partial class SafeContextTaskTest
+  {
+    [Flags]
+    public enum ContinueWithTestFlags
+    {
+      None = 0,
+      HasState = 1,
+      HasCancellationToken = 2,
+      HasTaskCompletionOptions = 4,
+      HasTaskScheduler = 8
+    }
+
+    public class ContinueWithTestContext<TTask, TContinuation>
+        where TTask : Task
+        where TContinuation : Delegate
+    {
+      public TTask Task { get; }
+
+      public TContinuation Continuation { get; }
+
+      public object State { get; }
+
+      public CancellationToken CancellationToken { get; }
+
+      public TaskContinuationOptions Options { get; }
+
+      public TaskScheduler TaskScheduler { get; }
+
+      public ContinueWithTestContext (TTask task, TContinuation continuation, object state, CancellationToken cancellationToken, TaskContinuationOptions options, TaskScheduler taskScheduler)
+      {
+        Task = task;
+        Continuation = continuation;
+        State = state;
+        CancellationToken = cancellationToken;
+        Options = options;
+        TaskScheduler = taskScheduler;
+      }
+    }
+
+    private class ContinueWithTestTaskScheduler : TaskScheduler
+    {
+      public bool WasCalled { get; set; }
+
+      protected override void QueueTask (Task task)
+      {
+        if ((task.CreationOptions & TaskCreationOptions.RunContinuationsAsynchronously) != 0)
+        {
+          WasCalled = true;
+          ThreadPool.QueueUserWorkItem(_ => { TryExecuteTask(task); });
+        }
+
+        // Synchronous calls will call TryExecuteTaskInline afterwards
+      }
+
+      protected override bool TryExecuteTaskInline (Task task, bool taskWasPreviouslyQueued)
+      {
+        if ((task.CreationOptions & TaskCreationOptions.RunContinuationsAsynchronously) == 0)
+        {
+          WasCalled = true;
+          return TryExecuteTask(task);
+        }
+
+        return false;
+      }
+
+      protected override IEnumerable<Task> GetScheduledTasks ()
+      {
+        throw new NotImplementedException();
```

Comment:
Even in tests, this should be NotSupportedException, not NotImplementedException.

The NIE is intended during development, not for completed types.